QStore4S/src/main/java/edu/asu/qstore4s/service/impl/RepositoryManager.java

```diff
@@ -178,6 +185,37 @@ public String getList(String xml, String accept) throws ParserException, IOExcep
     /**
      * {@inheritDoc}
      * 
+     * @throws ExecutionException
+     * 
+     * @throws JSONException
+     */
+    @Override
+    @Async
+    public void executeQueryAsync(String query, Class<?> clazz, String accept, Integer queryID)
+            throws ExecutionException {
+
+        asyncQueryManager.setQueryStatus(queryID, ExecutionStatus.RUNNING);
+
+        try {
+
+            List<CreationEvent> elementList = storeManager.executeQuery(query, clazz);
+
+            String res = "";
+            if (accept.equals(JSON)) {
+                res = converter.convertToJson(elementList);
+            } else {
+                res = converter.convertToXML(elementList);
+            }
```

Comment:
xml or json format should be created when requesting the results, not when creating them.